catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -1341,11 +1371,15 @@ public UpdateResponse update(UpdateStorageRequest streamUpdateRequest)
             if (updateStorageRequest != null) {
                 try {
                     storage.commit(updateStorageRequest);
+                    if (historianTransactionKey != null) {
+                        historian.commit(historianTransactionKey);
+                    }
                 } catch (StorageException e) {
                     LOGGER.error("Unable to commit content changes for id: "
                             + updateStorageRequest.getId(), e);
                     try {
                         storage.rollback(updateStorageRequest);
+                        historian.rollback(historianTransactionKey);
```

Comment:
Should wrap the call to `storage.rollback` in a try/catch so you can reliably call the `historian.rollback`.